backends/generators/generator.py

```diff
@@ -177,11 +273,27 @@ def load_instructions(
 
             encoding = data.get("encoding", {})
             if not encoding:
-                logging.error(
-                    f"Missing 'encoding' field in instruction {name} in {path}"
-                )
-                encoding_filtered += 1
-                continue
+                # Check if this instruction uses the new schema with a 'format' field
+                format_field = data.get("format")
+                if format_field:
+                    # Try to build a match string from the format field
+                    match_string = build_match_from_format(format_field)
+                    if match_string:
+                        # Create a synthetic encoding compatible with existing logic
+                        encoding = {"match": match_string, "variables": []}
+                        logging.debug(f"Built encoding from format field for {name}")
+                    else:
+                        logging.error(
+                            f"Could not build encoding from format field in instruction {name} in {path}"
+                        )
+                        encoding_filtered += 1
+                        continue
+                else:
+                    logging.error(
+                        f"Missing 'encoding' field in instruction {name} in {path}"
+                    )
+                    encoding_filtered += 1
+                    continue
```

Comment:
Let's swap the `if` and `else` cases to reduce the indentation:
```suggestion
                if not format_field:
                    logging.error(
                        f"Missing 'encoding' field in instruction {name} in {path}"
                    )
                    encoding_filtered += 1
                    continue

                # Try to build a match string from the format field
                match_string = build_match_from_format(format_field)
                if not match_string:
                    logging.error(
                        f"Could not build encoding from format field in instruction {name} in {path}"
                    )
                    encoding_filtered += 1
                    continue

                # Create a synthetic encoding compatible with existing logic
                encoding = {"match": match_string, "variables": []}
                logging.debug(f"Built encoding from format field for {name}")
```